### PR TITLE
Refactor Sponsorship

### PR DESCRIPTION
**Pull request by b264**
Fixes the build failure on develop

https://github.com/AgileVentures/osra/pull/442

### CODE REVIEW

**Comment by b264** [comment]
coverage: `99.181%` => `99.543%`

File: app/models/sponsorship.rb [hunk 1]
```diff
@@ -30,23 +30,12 @@ class Sponsorship < ActiveRecord::Base
 
 private
 
-  def start_date_no_later_than_1st_of_next_month
-    first_of_next_month = Date.current.beginning_of_month.next_month
-    if (self.start_date > first_of_next_month)
-      errors[:start_date] << "can not be later than the first of next month"
-    end
-  end
-
```
Comment (Maroo-b) [inline app/models/sponsorship.rb]: @b264 do you remember the utility of this private method ? I didn't find any references through the source code :thinking:

File: app/models/sponsorship.rb [hunk 2]
```diff
@@ -30,23 +30,12 @@ class Sponsorship < ActiveRecord::Base
 
 private
 
-  def start_date_no_later_than_1st_of_next_month
```
Comment (Maroo-b) [inline app/models/sponsorship.rb]: @b264 do you remember the utility of this private method ? I didn't find any references through the source code :thinking:

Comment (b264) [inline app/models/sponsorship.rb]: at some point it was replaced by the class https://github.com/AgileVentures/osra/blob/e5dc1e6c1f67f5eae8644df3d99a031e9a1b1449/app/models/validators/date_not_beyond_first_of_next_month_validator.rb

**Comment by Maroo-b** [comment]
+1

**Pull request merged**